src/silx/app/view/main.py

```diff
@@ -71,6 +73,39 @@ def createParser():
     return parser
 
 
+def filesArgToUrls(filenames: Iterable[str]) -> Generator[object, None, None]:
+    """Expand filenames and HDF5 data path in files input argument"""
+    import silx.io
+    from silx.io.utils import match
+    from silx.io.url import DataUrl
+    import silx.utils.files
+
+    for filename in filenames:
+        url = DataUrl(filename)
+
+        for file_path in sorted(silx.utils.files.expand_filenames([url.file_path()])):
+            if glob.has_magic(url.data_path()):
```

Comment:
will fail if `data_path()` is None must add something like 
```
if url.data_path() and glob.has_magic(url.data_path())
```

Reply:
Indeed

Reply:
Thanks for the testing! This is fixed.